src/components/Header/Header.jsx

```diff
@@ -24,14 +28,13 @@ const Header = ({ title }) => {
       <input
         className="city"
         type="text"
-        // value={city}
-        // onChange={(event) => setCity(event.target.value)}
+        value={city}
+        onChange={(event) => setCity(event.target.value)}
       />
 
-      <button>Search</button>
-
+      <button onClick={changeCity}>Search</button>
       <div>
-        <h1>Temp:{data.main.temp.toFixed()}°C</h1>
+        <h2>Temp:{temp.main.temp.toFixed()}°C</h2>
```

Comment:
Since you're saving **all** of the data, it might be misleading to call it `temp`. `Data` or `weather` or `weatherData` might be more appropriate.

Reply:
btw if I comment out line 37 then the app renders for me.

Reply:
Thanks, I have done the changes. yes, that's poor naming.